server/src/controllers/auth.controllers.ts

```diff
@@ -0,0 +1,262 @@
+
+    try {
+      const data = await verifyGoogleToken(client, tokenId);
+
+      if (data && data.email && data.name) {
+        const { email } = data;
+
+        const userRecord: UserDocument = await User.findOne({ email: email.toLowerCase() }).exec();
+
+        if (userRecord && userRecord.password !== '') {
+          const token = await generateJwt(
+            userRecord._id,
+            userRecord.email,
+            userRecord.accessType,
+            true
+          ); // always remember password for google login
+
+          sendResponse(res, 200, token);
+        } else {
+          sendResponse(res, 401, 'The provided email address is not registered');
+        }
+      }
+    } catch {
+      sendResponse(res, 500, 'Google token verification failed');
+    }
+  } else {
+    sendResponse(res, 422, 'Missing Token ID');
+  }
+};
+
+const signup: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const { ...toCreate }: UserSchema = req.body;
+
+  if (Object.keys(toCreate).length > 0 && toCreate.password && toCreate.email) {
+    try {
+      toCreate.email = toCreate.email.toLowerCase();
+      toCreate.password = await hashInput(toCreate.password);
+      const data: UserDocument = await User.create(toCreate);
+
+      if (data) {
+        sendResponse(res, 200, data);
+      } else {
+        sendResponse(res, 404, 'User account was not successfully created');
+      }
+    } catch (error: any) {
+      if (error?.name === 'ValidationError') {
+        sendResponse(res, 422, 'Missing required fields');
+      } else if (error?.code === 11000) {
+        sendResponse(res, 409, 'The provided email address is already in use');
+      } else {
+        sendResponse(res, 500, error?.message);
+      }
+    }
+  } else {
+    sendResponse(res, 422, 'Missing required fields');
+  }
+};
+
+const signupGoogle: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const { tokenId } = req.body;
+
+  if (tokenId) {
+    const client: OAuth2Client = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
+
+    try {
+      const data = await verifyGoogleToken(client, tokenId);
+
+      if (data && data.email && data.name) {
+        const { email, name: fullName } = data;
+        const newEmail = email.toLowerCase();
+
+        const userRecord: UserDocument = await User.findOne({ email: newEmail }).exec();
+
+        if (userRecord) {
+          sendResponse(res, 409, 'The provided email address is already in use');
+        }
+
+        const payload = { email: newEmail, fullName: fullName };
+
+        sendResponse(res, 200, payload);
+      }
+    } catch {
+      sendResponse(res, 500, 'Google token verification failed');
+    }
+  } else {
+    sendResponse(res, 422, 'Missing Token ID');
+  }
+};
+
+const forgotPassword: RequestHandler = async (req: Request, res: Response): Promise<void> => {
+  const { email } = req.body;
+
+  if (email) {
+    const newEmail = email.toLowerCase();
```

Comment:
if youre gonna do this here, then you might as well do the same at line 77 (i.e. assigning lowercase email to a variable). so you're consistent